### DISCUSSION TITLE
WPF BusinessListBase IsSavable/IsDirty

### DISCUSSION THREAD

**Discussion by CROprogrammer** [MarimerLLC/csla#3331] [Questions] [answered]
I'm using WPF and CSLA .NET 6.2.2. I want to use CSLA's IsSavable and IsDirty from BusinessListBase class type in WPF to enable/disable button but it does not work. It does work for BusinessBase, but not for BusinessListBase. Why is that?
I've changed its elements in WPF ui, so I'm sure that at least one of those elements is dirty and savable, and I'm expecting that list should then be dirty and savable.

**Comment by rockfordlhotka** [comment, marked as answer]
iirc, this is because `BusinessListBase` (BLB) doesn't raise `PropertyChanged` events like `BusinessBase` (BB). 

The historical reason why BLB doesn't have managed properties or other things like BB is that Windows Forms and Web Forms didn't support binding to properties of a list. It wasn't until WPF that such a thing was possible.

There's actually a backlog item to make BLB have all the BB capabilities: #1830

I _think_ you may be able to work around this though, if you handle the `ChildChanged` event in your list class, and call `PropertyHasChanged("IsSavable")` and `PropertyHasChanged("IsDirty")` when a child changes.

That said, I'm a bit surprised that this isn't automatic - but it may be for the reasons I've listed here.
